ddtrace/contrib/dbapi/__init__.py

```diff
@@ -73,16 +122,36 @@ class TracedConnection(wrapt.ObjectProxy):
     def __init__(self, conn, pin=None):
         super(TracedConnection, self).__init__(conn)
         name = _get_vendor(conn)
+        self._self_datadog_name = '{}.connection'.format(name)
         db_pin = pin or Pin(service=name, app=name, app_type=AppTypes.db)
         db_pin.onto(self)
 
+    def _trace_method(self, method, name, extra_tags, *args, **kwargs):
+        pin = Pin.get_from(self)
+        if not pin or not pin.enabled():
+            return method(*args, **kwargs)
+        service = pin.service
+
+        with pin.tracer.trace(name, service=service) as s:
+            s.set_tags(pin.tags)
+            s.set_tags(extra_tags)
+
+            return method(*args, **kwargs)
+
     def cursor(self, *args, **kwargs):
         cursor = self.__wrapped__.cursor(*args, **kwargs)
         pin = Pin.get_from(self)
         if not pin:
             return cursor
         return TracedCursor(cursor, pin)
 
+    def commit(self, *args, **kwargs):
+        span_name = '{}.{}'.format(self._self_datadog_name, 'commit')
+        self._trace_method(self.__wrapped__.commit, span_name, {}, *args, **kwargs)
```

Comment:
Should these being `return self._trace_method(...)`?

Is this going to cause any problems?

Reply:
Perhaps not because I don't think the return value is used from these methods... but we definitely should be returning. Great catch.

Reply:
yes, it does appear that they both return none, but adding a `return ` to them isn't going to hurt, and will make sure if a specific implementation _does_ return something, that we don't break anything.

Should be harmless either way.